PlainTasks.py

```diff
@@ -833,3 +842,146 @@ def run(self, edit):
+        if success is True:
+            self.view.erase(edit,region)
+
+        return
+
+    def __writeArchive(self, filename, start_line, end_line):
+        # Write out the given line numbers to a file.
+        # Since we need it to be in a region to delete, anyway, 
+        # go ahead and create the region and return it.
+
+        # Build our region
+        start_region=self.view.line(self.view.text_point(start_line,0))
+        region=start_region.cover(self.view.line(
+            self.view.text_point(end_line,0)))
+
+        # Write it out!
+        sublime.status_message('Archiving tree to {}'.format(filename))
+        try:
+            # Have to use io.open because windows doesn't like writing
+            # utf8 to regular filehandles
+            with io.open(filename, 'a', encoding='utf8') as fh:
+                data = self.view.substr(region)
+                # Is there a way to read this in?
+                fh.write("--- ✄ -----------------------\n")
+                fh.write("Archived {}:\n".format(datetime.now().strftime(
+                    self.date_format)))
+                # And, finally, write our data
+                fh.write("{}\n".format(data))
+            return True, region
+
+        except Exception as e:
+            sublime.error_message("Error:\n\nUnable to append to {}\n{}".format(
+                filename, str(e)))
+            return False, None
+
+    def __createArchiveFilename(self):
+        # Create our archive filename, from the mask in our settings.
+
+        # Split filename int dir, base, and extension, then apply our mask
+        path_base, extension=os.path.splitext(self.view.file_name())
+        dir=os.path.dirname(path_base)
+        base=os.path.basename(path_base)
+        sep=os.sep
+
+        # Now build our new filename
+        try:
+            # This could fail, if someone messed up the mask in the
+            # settings.  So, if it did fail, use our default.
+
+            archive_filename=self.archive_org_filemask.format(
+                dir=dir, base=base, ext=extension, sep=sep)
+        except:
+            # Use our default mask
+            archive_filename=self.archive_org_default_filemask.format(
+                    dir=dir, base=base, ext=extension, sep=sep)
+
+            # Display error, letting the user know
+            sublime.error_message("Error:\n\nInvalid filemask:{}\nUsing default: {}".format(
+                self.archive_org_filemask, self.archive_org_default_filemask))
+
+        return archive_filename
+
+    def __regionIndentLen(self, region):
+        # Compute how many spaces a given region is indented
+        # This is used to determine when we've left our subtree
+
+        line_contents  = self.view.substr(region).rstrip()
+        indent = re.match('^(\s*)\S', line_contents, re.U)
+        if indent is None or indent.group(1) is None:
+            return 0
+        return len(indent.group(1))
+
+    def __findCurrentSubtree(self):
```

Comment:
No, there is two different methods `view.full_line` and `view.line`.
For full line which contains line break the end is actually the beginning of next line (unless it is eof) which is exactly what we need in this case. 
For line without line break the end is the actual end so results is the same.